subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/MultiVersionIntegrationSpec.groovy

```diff
@@ -29,6 +29,8 @@ abstract class MultiVersionIntegrationSpec extends AbstractIntegrationSpec {
     static def version
 
     static VersionNumber getVersionNumber() {
-        VersionNumber.parse(version.toString())
+        // Split off a potential classifier
+        def m = version.toString() =~ /^(.*?)(?::.*)?$/
```

Comment:
Is the idea here to get the last index of `:` in the version string and then take everything before that? I think using something like `String.indexOf()` or `String.lastIndexOf()` would be better readable than this regex - at least I have a hard time understanding it.

Reply:
It cuts at the first `:`, but that's what it does. Do you have a proposal for the code to replace this with? To me this looks pretty simple and easy to understand already, especially given the comment.

Reply:
I suppose this should do the trick:

```
String versionString = version.toString()
int firstColonIndex = versionString.indexOf(':')
String versionPart = firstColon == -1 ? versionString : versionString.substring(0, firstColonIndex)
```

This version is simpler to understand for me.